src/transaction.ts

```diff
@@ -225,7 +402,22 @@ export const sendTransaction = async function (
  * @returns An object with transaction IDs, transactions, group transaction ID (`groupTransactionId`) if more than 1 transaction sent, and (if `skipWaiting` is `false` or unset) confirmation (`confirmation`)
  */
 export const sendAtomicTransactionComposer = async function (atcSend: AtomicTransactionComposerToSend, algod: Algodv2) {
-  const { atc, sendParams } = atcSend
+  const { atc: givenAtc, sendParams } = atcSend
+
+  let atc: AtomicTransactionComposer
+
+  const hasAppCalls = () =>
+    givenAtc
+      .buildGroup()
+      .map((t) => t.txn.type)
+      .includes(algosdk.TransactionType.appl)
+
+  // If packResources is true OR if packResources is undefined and there are app calls, then pack resources
+  if (sendParams?.packResources || (sendParams?.packResources === undefined && hasAppCalls())) {
```

Comment:
If it wasn't for this (adding the call to `sendAtomicTransactionComposer`, then I'd suggest adding `packResources` to `CoreAppCallArgs` so it's explicitly tied to app call transactions, but then we can't do this here so on balance `sendParams` is probably the right place for it.